tests/keras/metrics_test.py

```diff
@@ -100,5 +106,76 @@ def test_sparse_top_k_categorical_accuracy():
     assert failure_result == 0
 
 
+@keras_test
+def test_stateful_metrics():
+    np.random.seed(1334)
+
+    class BinaryTruePositives(keras.layers.Layer):
+        """Stateful Metric to count the total true positives over all batches.
+
+        Assumes predictions and targets of shape `(samples, 1)`.
+
+        # Arguments
+            threshold: Float, lower limit on prediction value that counts as a
+                positive class prediction.
+            name: String, name for the metric.
+        """
+
+        def __init__(self, name='true_positives', **kwargs):
```

Comment:
I suggest it be possible to define the y_true y_pred shapes here. In particular this applies to multiple hypothesis networks.